chpl/chpl-service/src/main/java/gov/healthit/chpl/scheduler/job/summarystatistics/SummaryStatisticsCreatorJob.java

```diff
@@ -61,51 +74,71 @@ public SummaryStatisticsCreatorJob() throws Exception {
+            LOGGER.info("Completing getting all listings.");
 
-            EmailStatistics emailBodyStats = emailStatisticsCreator.getStatistics();
+            EmailStatistics emailBodyStats = emailStatisticsCreator.getStatistics(allListings);
+            saveSummaryStatistics(emailBodyStats);
 
+            Boolean generateCsv = Boolean.valueOf(jobContext.getMergedJobDataMap().getString("generateCsvFile"));
             if (generateCsv) {
-                createSummaryStatisticsFile(startDate, endDate, numDaysInPeriod);
+                createSummaryStatisticsFile(allListings, jobContext);
             }
-            saveSummaryStatistics(emailBodyStats, endDate);
-
         } catch (Exception e) {
             LOGGER.error("Caught unexpected exception: " + e.getMessage(), e);
         }
         LOGGER.info("********* Completed the Summary Statistics Creation job. *********");
     }
 
-    private void createSummaryStatisticsFile(final Date startDate, final Date endDate, final Integer numDaysInPeriod)
+
+    @SuppressWarnings("checkstyle:linelength")
+    private void createSummaryStatisticsFile(List<CertifiedProductDetailsDTO> allListings, JobExecutionContext jobContext)
             throws InterruptedException, ExecutionException {
+
+
+        if (!isGenerateStatisticsFlagOn(jobContext)) {
+            return;
+        }
+
+        Date startDate = getStartDate();
+        if (startDate == null) {
+            throw new RuntimeException("Could not obtain the startDate.");
+        }
+        Date endDate = new Date();
+        Integer numDaysInPeriod = Integer.valueOf(env.getProperty("summaryEmailPeriodInDays").toString());
+
+
         List<CsvStatistics> csvStats = new ArrayList<CsvStatistics>();
         Calendar startDateCal = Calendar.getInstance(TimeZone.getTimeZone(ZoneOffset.UTC));
         startDateCal.setTime(startDate);
         Calendar endDateCal = Calendar.getInstance(TimeZone.getTimeZone(ZoneOffset.UTC));
         endDateCal.setTime(startDate);
         endDateCal.add(Calendar.DATE, numDaysInPeriod);
 
+        Map<Long, List<CertificationStatusEventDTO>> statusesForAllListings = getAllStatusesForAllListings();
+
         while (endDate.compareTo(endDateCal.getTime()) >= 0) {
             LOGGER.info("Getting csvRecord for start date " + startDateCal.getTime().toString() + " end date "
                     + endDateCal.getTime().toString());
+
             DateRange csvRange = new DateRange(startDateCal.getTime(), new Date(endDateCal.getTimeInMillis()));
             CsvStatistics historyStat = new CsvStatistics();
+
             historyStat.setDateRange(csvRange);
-            historyStat = historicalStatisticsCreator.getStatistics(csvRange);
+            historyStat = historicalStatisticsCreator.getStatistics(allListings, statusesForAllListings, csvRange);
+
             csvStats.add(historyStat);
+
             LOGGER.info("Finished getting csvRecord for start date " + startDateCal.getTime().toString() + " end date "
```

Comment:
As far as I can tell, the only place the `startDate` part of the `dateRange` variable is used is in this logging statement. Even in the initial "set up the dates" thing, it's not using the dateRange variable yet, just the response from the `getStartDate()` method. Is it worth keeping the dateRange around for just this statement? The log could be written instead as something like `Finished getting records for <numDaysInPeriod> ending at <endDate>` and it would have the same result